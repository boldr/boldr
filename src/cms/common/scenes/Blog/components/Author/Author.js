/* @flow */
import React from 'react';
import { Card, Feed } from 'semantic-ui-react';

const Author = (props: { display_name: String, avatar_url: String }) => {
  return (
    <div className="post__sidebar-author">
    <Card className="post__sidebar-card">
      <Card.Content>
        <Feed>
          <Feed.Event>
            <Feed.Label image={ props.avatar_url } />
            <Feed.Content>
              <Feed.Summary>
                { props.display_name }
              </Feed.Summary>
            </Feed.Content>
          </Feed.Event>
        </Feed>
      </Card.Content>
    </Card>
      </div>
  );
};

export default Author;
