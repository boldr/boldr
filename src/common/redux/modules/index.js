import { combineReducers } from 'redux';
import { routerReducer as routing } from 'react-router-redux';
import auth from './auth/auth.reducer';

const rootReducer = combineReducers({
  routing,
  auth
});

export default rootReducer;
