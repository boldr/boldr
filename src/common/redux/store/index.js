let store;

if (__DEV__) {
  store = require('./configureStore.dev').default;
} else {
  store = require('./configureStore.prod').default;
}

export default store;
