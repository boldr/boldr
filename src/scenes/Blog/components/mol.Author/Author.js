import React from 'react';
import Avatar from '../../../../components/md/Avatars';
import { Heading } from '../../../../components';

const styles = {
  wrapper: {
    display: 'flex',
    flexWrap: 'wrap',
    alignItems: 'center'
  }
};
const Author = (props) => {
  return (
    <div style={ styles.wrapper }>
      <Avatar src={ props.avatarUrl } />
      <Heading size={ 3 }> { props.displayName }</Heading>
    </div>
  );
};

Author.propTypes = {
  displayName: React.PropTypes.string.isRequired,
  avatarUrl: React.PropTypes.string
};

export default Author;
