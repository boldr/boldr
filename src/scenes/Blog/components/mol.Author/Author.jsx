import React from 'react';

const Author = (props) => {
  return (
    <div>
      { props.user.displayName }
    </div>
  );
};

Author.propTypes = {
  displayName: React.PropTypes.string.isRequired
};

export default Author;
