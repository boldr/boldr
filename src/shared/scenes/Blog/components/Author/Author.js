/* @flow */
import React from 'react';
import styled from 'styled-components';
import classnames from 'classnames';
import Link from 'react-router-dom/Link';
import {Avatar, Heading, Col, Social, StyleClasses} from 'boldr-ui';

const BASE_ELEMENT = StyleClasses.POST_SIDEBAR_AUTHOR;

const AvatarName = styled.div`
  display: inline-block;
  vertical-align: middle;
  &:first-child {
    margin-right: 15px;
  }
`;
const BioBlock = styled.div`
  vertical-align: middle;
  margin-bottom: 15px;
  padding-top: 10px;
`;

type Props = {
  className: string,
  username: string,
  avatarUrl: string,
  lastName: string,
  firstName: string,
  bio: string,
  social: Object,
};
const Author = (props: Props) => {
  const classes = classnames(BASE_ELEMENT, props.className);
  const authorName = (
    <Link to={`/profiles/${props.username}`}>{props.username}</Link>
  );
  return (
    <div className={classes}>
      <AvatarName>
        <Avatar src={props.avatarUrl} role="presentation" />
      </AvatarName>
      <AvatarName><Heading size={3}>{authorName}</Heading></AvatarName>
      <BioBlock>{props.bio}</BioBlock>
      <Social
        facebook
        fburl={props.social.facebook.url}
        twitter
        turl={props.social.twitter.url}
        google
        gurl={props.social.google.url}
        github
        ghurl={props.social.github.url}
        linkedin
        lurl={props.social.linkedin.url}
      />
    </div>
  );
};

Author.defaultProps = {
  avatarUrl: '',
};

export default Author;
