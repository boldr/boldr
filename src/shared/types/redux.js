/* @flow */
import type { Store as ReduxStore } from 'redux';

export type ApiAction<T> = {
  key: string,
  payload: T,
};

export type ActionType =
  | 'LOGIN_REQUEST'
  | 'LOGIN_SUCCESS'
  | 'LOGIN_FAILURE'
  | 'LOGOUT_USER'
  | 'LOGOUT_USER_FAIL'
  | 'CHECK_AUTH_REQUEST'
  | 'CHECK_AUTH_SUCCESS'
  | 'CHECK_AUTH_FAILURE'
  | 'SIGNUP_USER_REQUEST'
  | 'SIGNUP_USER_SUCCESS'
  | 'SIGNUP_USER_FAILURE'
  | 'FORGOT_PASSWORD_REQUEST'
  | 'FORGOT_PASSWORD_SUCCESS'
  | 'FORGOT_PASSWORD_FAILURE'
  | 'RESET_PASSWORD_REQUEST'
  | 'RESET_PASSWORD_SUCCESS'
  | 'RESET_PASSWORD_FAILURE'
  | 'LOAD_POSTS_REQUEST'
  | 'LOAD_POSTS_SUCCESS'
  | 'LOAD_POSTS_FAILURE'
  | 'FETCH_POST_REQUEST'
  | 'FETCH_POST_SUCCESS'
  | 'FETCH_POST_FAILURE'
  | 'UPDATE_POST_REQUEST'
  | 'UPDATE_POST_SUCCESS'
  | 'UPDATE_POST_FAILURE'
  | 'SELECT_POST'
  | 'SELECT_POST_SUCCESS'
  | 'SELECT_POST_FAILURE'
  | 'CREATE_POST_REQUEST'
  | 'CREATE_POST_SUCCESS'
  | 'CREATE_POST_FAILURE'
  | 'DELETE_POST_FAILURE'
  | 'DELETE_POST_REQUEST'
  | 'DELETE_POST_SUCCESS';

export type PostsReducer = {
  all: Object,
  ids: Array<Object>,
  isFetching: boolean,
  currentPost: Object,
};

export type BlogReducer = {
  posts: PostsReducer,
  tags: any,
  comments: any,
};

export type UserInfo = {
  [userId: string]: {
    readyStatus: string,
    err: any,
    info: Object,
  },
};

export type Reducer = {
  routing: any,
  blog: BlogReducer,
};


export type Action =
  { type: '@boldr/blog/FETCH_POSTS_REQUEST' } |
  { type: '@boldr/blog/FETCH_POSTS_SUCCESS', data: Array<Object> } |
  { type: '@boldr/blog/FETCH_POSTS_FAILURE', err: any } |
  { type: '@boldr/blog/FETCH_POST_REQUEST', userId: string } |
  { type: '@boldr/blog/FETCH_POST_SUCCESS', userId: string, data: Object } |
  { type: '@boldr/blog/FETCH_POST_FAILURE', userId: string, err: any };

export type Store = ReduxStore<Reducer, Action>;
// eslint-disable-next-line no-use-before-define
export type Dispatch = (action: Action | ThunkAction | PromiseAction | Array<Action>) => any;
export type GetState = () => Object;
export type ThunkAction = (dispatch: Dispatch, getState: GetState) => any;
export type PromiseAction = Promise<Action>;
