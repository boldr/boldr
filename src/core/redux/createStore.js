import { createStore as _createStore, applyMiddleware, compose } from 'redux';
import { routerMiddleware } from 'react-router-redux';
import thunkMiddleware from 'redux-thunk';
import createLogger from 'redux-logger';
import rootReducer from '../../state';
import createMiddleware from './clientMiddleware';

const ISDEV = process.env.NODE_ENV === 'development';
export default function createStore(history, client, data) {
  // Sync dispatched route actions to the history
  const reduxRouterMiddleware = routerMiddleware(history);
  const logger = createLogger();
  const middleware = [thunkMiddleware, createMiddleware(client), reduxRouterMiddleware, logger];

  let finalCreateStore;
  if (ISDEV) {
    finalCreateStore = compose(
      applyMiddleware(...middleware),
      typeof window === 'object' && typeof window.devToolsExtension !== 'undefined' ? window.devToolsExtension() : f => f // eslint-disable-line
    )(_createStore);
  } else {
    finalCreateStore = applyMiddleware(...middleware)(_createStore);
  }

  const store = finalCreateStore(rootReducer, data);


  if (ISDEV && module.hot) {
    module.hot.accept('../../state', () => {
      const nextReducer = require('../../state');

      store.replaceReducer(nextReducer);
    });
  }

  return store;
}
