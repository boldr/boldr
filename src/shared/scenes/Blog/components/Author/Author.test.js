import React from 'react';
import { createStore } from 'redux';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Author from './Author';

const authorProps = {
  avatarUrl: 'abcd',
  username: 'hey',
  firstName: 'me',
  lastName: 'you',
  bio: 'bio bio',
  social: {
    facebook: {
      url: 'asf',
    },
    twitter: {
      url: 'asf',
    },
    google: {
      url: 'asf',
    },
    github: {
      url: 'asf',
    },
    linkedin: {
      url: 'asf',
    },
  },
};

describe('<Author />', () => {
  const wrapper = shallow(<Author {...authorProps} />);
  it('renders <Author /> without breaking', () => {
    expect(wrapper.find('div').length).toBe(1);
    expect(wrapper.instance().props.username).toBe('hey');
  });
  it('renders snapshot', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
