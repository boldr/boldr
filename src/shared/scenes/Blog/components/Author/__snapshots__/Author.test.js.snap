// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Author /> renders snapshot 1`] = `
<div
  className="boldrui-sidebar-author"
>
  <styled.div>
    <Avatar
      role="presentation"
      src="abcd"
      suffixes={
        Array [
          "red",
          "pink",
          "purple",
          "deep-purple",
          "indigo",
          "blue",
          "light-blue",
          "cyan",
          "teal",
          "green",
          "light-green",
          "lime",
          "yellow",
          "amber",
          "orange",
          "deep-orange",
          "brown",
          "grey",
          "blue-grey",
        ]
      }
    />
  </styled.div>
  <styled.div>
    <Heading
      size={3}
    >
      <Link
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/profiles/hey"
      >
        hey
      </Link>
    </Heading>
  </styled.div>
  <styled.div>
    bio bio
  </styled.div>
  <Social
    facebook={true}
    fburl="asf"
    ghurl="asf"
    github={true}
    google={true}
    gurl="asf"
    linkedin={true}
    lurl="asf"
    turl="asf"
    twitter={true}
  />
</div>
`;
